src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1058,27 +1246,44 @@ public void ProcessFileChange(string fullFilePath)
                         // all adds/removals of documents for this project happen under our lock, it's safe to do this
                         // check without taking the main workspace lock
                         var document = _project._workspace.CurrentSolution.GetDocument(documentId);
-
                         if (document == null)
                         {
                             return;
                         }
 
+                        _project._documentIdToDynamicFileInfoProvider.TryGetValue(documentId, out var fileInfoProvider);
+
                         _project._workspace.ApplyChangeToWorkspace(w =>
                         {
                             if (w.IsDocumentOpen(documentId))
                             {
                                 return;
                             }
 
-                            var documentInfo =
-                                DocumentInfo.Create(
-                                    document.Id,
-                                    document.Name,
-                                    document.Folders,
-                                    document.SourceCodeKind,
-                                    new FileTextLoader(fullFilePath, defaultEncoding: null),
-                                    document.FilePath);
+                            TextLoader textLoader;
+                            IDocumentServiceProvider documentServiceProvider;
+                            if (fileInfoProvider == null)
+                            {
+                                textLoader = new FileTextLoader(fullFilePath, defaultEncoding: null);
+                                documentServiceProvider = null;
+                            }
+                            else
+                            {
+                                var fileInfo = _project._threadingContext.JoinableTaskFactory.Run(() => fileInfoProvider.GetDynamicFileInfoAsync(_project.Id, _project._filePath, fullFilePath, CancellationToken.None));
```

Comment:
@heejaechang  - this isn't right. 

This is passing in the mangled file path. Ex:
C:\Users\rynowak\Source\Repos\WebApplication6\WebApplication6\Views\Shared\Error.cshtml#Generated#
instead of 
C:\Users\rynowak\Source\Repos\WebApplication6\WebApplication6\Views\Shared\Error.cshtml

Reply:
oops. my bad. I will make it to give you right path.

Reply:
fixed.